### PR TITLE
Configurable array length, number values, and response conditions

### PR DESCRIPTION
**Pull request by senning**
Hi there! Thanks for building this library, I'm finding it really helpful for quick prototyping. I ran into a few pain points that I had to resolve by directly changing the mock file, so I tried to fix them with this PR. There are two groups of changes, one of which is much bigger, so I kept them in separate commits in case you wanted to cherry-pick the smaller one.

### [support min and max values for numbers and array length](https://github.com/zoubingwu/msw-auto-mock/commit/34e3597311bd356d2a3a7916fc2725ffd1196c45)

- Use the `minItems` and `maxItems` properties in array types to set the generated length of the array (replacing the default `1 - MAX_ARRAY_LENGTH`)
- Use the `minimum` and `maximum` properties for number types to populate the `min` and `max` props

### [alt handler transformer for conditional responses](https://github.com/zoubingwu/msw-auto-mock/commit/11fa5a144eabc3f861e87014ebef8ed6b7002d6b)

- add an option for a `responseConditions` file which allows specifying the conditions under which to use specific response handlers. If the file specifies handlers for a route + verb combo, it replaces the resultArray handler
- with this refactor, the [default response](https://swagger.io/docs/specification/describing-responses/#default) can also be supported

To provide an example of these changes in action, I set up [this repo](https://github.com/senning/msw-auto-mock-demo). You can run it with these params:

```
./test.yml -o ./mock.js -r ./responseConditions.json --node --base-url https://test.com/
```

and test it with the tests in that repo.

### CODE REVIEW

**Comment by zoubingwu** [comment]
Thanks for your contribution! Could you please use pnpm as package manager? or just remove package-lock.json from commits

File: src/transform.ts [hunk 1]
```diff
@@ -116,9 +147,9 @@ function transformJSONSchemaToFakerCode(jsonSchema?: OpenAPIV3.SchemaObject, key
           .join(',\n')}
     }`;
     case 'array':
-      return `[...(new Array(faker.datatype.number({ min: ${jsonSchema.minLength ?? 1}, max: ${
-        jsonSchema.maxLength ?? 'MAX_ARRAY_LENGTH'
-      } }))).keys()].map(_ => (${transformJSONSchemaToFakerCode(jsonSchema.items as OpenAPIV3.SchemaObject)}))`;
+      return `[...(new Array(faker.datatype.number({ min: ${jsonSchema.minItems ?? 1}, max: ${jsonSchema.maxItems ?? 'MAX_ARRAY_LENGTH'} }))).keys()].map(_ => (${transformJSONSchemaToFakerCode(
```
Comment (senning) [inline src/transform.ts:150]: Array length limits should be specified with `minItems` and `maxItems`. `*length` properties only apply to [strings](https://datatracker.ietf.org/doc/html/draft-wright-json-schema-validation-00#section-4.1).

File: src/transform.ts [hunk 2]
```diff
@@ -98,7 +129,7 @@ function transformJSONSchemaToFakerCode(jsonSchema?: OpenAPIV3.SchemaObject, key
       return transformStringBasedOnFormat(jsonSchema.format, key);
     case 'number':
     case 'integer':
-      return `faker.datatype.number({ min: ${jsonSchema.minimum}, max: ${jsonSchema.maximum} })`;
+      return transformNumberBasedOnSchema(jsonSchema);
```
Comment (senning) [inline src/transform.ts:132]: `transformNumberBasedOnSchema` creates an object for the min/max values and runs it through `JSON.stringify`. This strips out undefined values, keeping the mock cleaner.

**Comment by leogallot** [comment]
Hello, is this PR planned to be merged? Thank you.

**Pull request closed without merging**